src/python_testing/TC_EWATERHTR_2_1.py

```diff
@@ -0,0 +1,96 @@
+# for details about the block below.
+#
+# === BEGIN CI TEST ARGUMENTS ===# test-runner-runs: run1
+# test-runner-run/run1/app: ${WATER_HEATER_MANAGEMENT_APP}
+# test-runner-run/run1/factoryreset: True
+# test-runner-run/run1/quiet: True
+# test-runner-run/run1/app-args: --discriminator 1234 --KVS kvs1 --trace-to json:${TRACE_APP}.json --enable-key 000102030405060708090a0b0c0d0e0f --featureSet 0x03
+# test-runner-run/run1/script-args: --storage-path admin_storage.json --commissioning-method on-network --discriminator 1234 --passcode 20202021 --hex-arg enableKey:000102030405060708090a0b0c0d0e0f --endpoint 1 --trace-to json:${TRACE_TEST_JSON}.json --trace-to perfetto:${TRACE_TEST_PERFETTO}.perfetto
+# === END CI TEST ARGUMENTS ===
+
+import logging
+
+import chip.clusters as Clusters
+from TC_EWATERHTRBase import EWATERHTRBase
+from matter_testing_support import MatterBaseTest, TestStep, async_test_body, default_matter_test_main
+
+logger = logging.getLogger(__name__)
+
+
+class TC_EWATERHTR_2_1(MatterBaseTest, EWATERHTRBase):
+
+    def desc_TC_EWATERHTR_2_1(self) -> str:
+        """Returns a description of this test"""
+        return "[TC-EWATERHTR-2.1] Attributes with attributes with DUT as Server\n" \
+            "This test case verifies the non-global attributes of the Water Heater Management cluster server."
+
+    def pics_TC_EWATERHTR_2_1(self):
+        """ This function returns a list of PICS for this test case that must be True for the test to be run"""
+        return ["EWATERHTR.S", "EWATERHTR.S.F00", "EWATERHTR.S.F01"]
+
+    def steps_TC_EWATERHTR_2_1(self) -> list[TestStep]:
+        steps = [
+            TestStep("1", "Commissioning, already done",
+                     is_commissioning=True),
+            TestStep("2", "TH reads HeaterTypes attribute.",
+                     "DUT as Server replies with a WaterHeaterTypeBitmap (enum8)greater than 0x00 (at least one type supported), and less than 0x20 (no undefined types supported)."),
+            TestStep("3", "TH reads HeatDemand attribute.",
+                     "DUT as Server replies with a WaterHeaterDemandBitmap (enum8)."),
+            TestStep("4", "TH reads TankVolume attribute.",
+                     "DUT as Server replies with a uint16 value."),
+            TestStep("5", "TH reads EstimatedHeatRequired attribute.",
+                     "DUT as Server replies with an energy-mWh value."),
+            TestStep("6", "TH reads TankPercentage attribute.",
+                     "DUT as Server replies with a percent value."),
+            TestStep("7", "TH reads BoostState attribute.",
+                     "DUT as Server replies with a BoostStateEnum (enum8) value."),
+        ]
+
+        return steps
+
+    @async_test_body
+    async def test_TC_EWATERHTR_2_1(self):
+
+        self.step("1")
+        # Commission DUT - already done
+
+        # Note the values used here are configured in WhmManufacturer::Init()
```

Comment:
Remove comment - since we don't care about the example app implementation here.

Reply:
Comment removed